lms/static/js/edxnotes/utils/notes_collector.js

```diff
@@ -0,0 +1,87 @@
+(function(define) {
+    'use strict';
+    define(['jquery', 'underscore', 'annotator_1.2.9'], function($, _, Annotator) {
+        var cleanup,
+            renderNotes,
+            fetchNotesWhenReady,
+            storeNotesRequestData,
+            searchRequestsData = [];
+
+        cleanup = function () {
+            searchRequestsData = [];
+        };
+
+        storeNotesRequestData = function (data) {
+            searchRequestsData.push(data);
+            fetchNotesWhenReady();
+        };
+
+        fetchNotesWhenReady = function () {
+            if ($('.edx-notes-wrapper').length != searchRequestsData.length) {
```

Comment:
I think it makes sense, because it is called every time inside this method instead. I think making the code separation cleaner can only help with maintenance.

Note that there is an interesting question as to what should happen when the count changes. Should this then be assumed to be the first rendering of a fresh set of blocks? Or will it be the case that the collector's cleanUp method will be called first in that case?